charts/mlp/templates/deployment.yaml

```diff
@@ -24,10 +24,13 @@ spec:
       labels:
         app: {{ template "mlp.name" .}}
         release: {{ .Release.Name }}
+        {{- if .Values.mlp.podLabels }}
+        {{- toYaml .Values.mlp.podLabels | nindent 8 }}
+        {{- end }}
```

Comment:
Should we also add this in metadata.labels above (line 6)?

Reply:
`metadata.labels` != `spec.template.metadata.labels`
The former is applied to the `Deployment`, where the latter is applied to `Pod`. Since the config key is `podLabels`, it should only be applied to pods. 

It's possible to introduce `deploymentLabels` and to apply them to the `Deployment` (ref: https://github.com/istio/istio/blob/d9b722d7cf5b32756263c81faeb3f8e9ebfe59fa/manifests/charts/istio-control/istio-discovery/values.yaml#L58), but it's not necessary for now.

Reply:
Oh, I don't think we need it. But if we want to allow the user to add their custom label for deployment resource, we should provide `deploymentLabels` value instead. Something like Prometheus did: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/server/deploy.yaml#L31-L34

Reply:
out of scope for now :) This chart requires a lot of polishing anyway